Assets/Scripts/CombatManager.cs

```diff
@@ -2,78 +2,78 @@
     }
 
-    void GenerateCreature(GameObject creaturePrefab) {
-        GameObject creature = Instantiate(creaturePrefab);
+    void GenerateCreature (GameObject creaturePrefab) {
+        GameObject creature = Instantiate (creaturePrefab);
 
-        creature.transform.SetParent(gameObject.transform, false);
+        creature.transform.SetParent (gameObject.transform, false);
 
-        creature.GetComponent<CreatureController>().combatManager = this;
+        creature.GetComponent<CreatureController> ().combatManager = this;
     }
 
-    public void HandleCreatureClick(CreatureController creature) {
-
+    public void HandleCreatureClick (CreatureController creature) {
         // Check for target selection and update creature displays accordingly
-        if (targets.Contains(creature)) {
-            targets.Remove(creature);
-
-            creature.SetTargetingArrowDisplay(false);
-
+        if (targets.Contains (creature)) {
+            targets.Remove (creature);
+            creature.SetTargetingArrowDisplay (false);
         } else {
-            targets.Add(creature);
-
-            creature.SetTargetingArrowDisplay(true);
-
+            targets.Add (creature);
+            creature.SetTargetingArrowDisplay (true);
         }
-
         // Update UI if all conditions for casting a spell have been met
-        CheckCastReady();
+        CheckCastReady ();
     }
 
-    public bool ParseSpell(int numCards) {
+    public bool IsSpellValid (List<CardController> selectedCards) {
         // TODO: return true and set ref if cards selected match a spell in database
-        
-        this.cardsSelected = numCards;
+        this.selectedCards = selectedCards;
 
         // Update UI if all conditions for casting a spell have been met
-        CheckCastReady();
-
+        CheckCastReady ();
+        return spellProcessor.IsSpellComplete (selectedCards);
         // Let caller know if card selection conditions have been met
-        return this.cardsSelected == cardsRequired;
     }
 
     void CheckCastReady () {
-        castButton.SetReady(targets.Count != 0 && cardsSelected == cardsRequired);
+        castButton.SetReady (targets.Count != 0 && selectedCards.Count == cardsRequired);
     }
 
-    public void ExecuteSpell() {
-        // TODO: Apply the spell effects on the target
+    public void ExecuteSpell () {
+        // Compiling a list of all creatures that have died after the spell
+        List<CreatureController> targetsToRemove = new List<CreatureController> ();
+        spellProcessor.CastSpell (selectedCards, targets);
+        foreach (CreatureController creature in targets) {
+            if (creature.isCreatureDead ()) {
+                creature.DestroyCreature ();
+                targetsToRemove.Add (creature);
+            }
+        }
+        // Requires a predicate for RemoveAll call so I'm just using a lambda expression
+        targets.RemoveAll (target => targetsToRemove.Contains (target));
```

Comment:
Could get rid of the `targetsToRemove` list if you did:

```suggestion
        targets.RemoveAll (target => target.isCreatureDead());
```

Reply:
I get why you didn't, because the target GameObject should be destroyed. Turns out `Object.Destroy()` takes effect after the current frame, so the refs of that object stick around for the duration of the current synchronous call stack!

From the Unity docs for `Object.Destroy`:
"Actual object destruction is always delayed until after the current Update loop, but is always done before rendering."